Unit test TestRamStoreWatchTimeout is flaky

**Describe the bug**
The unit test failed occasionally:
```
--- FAIL: TestRamStoreWatchTimeout (0.14s)
    store_test.go:509: w2 was not stopped, expected stopped
    store_test.go:511:
                Error Trace:    store_test.go:511
                Error:          Not equal:
                                expected: 1
                                actual  : 2
                Test:           TestRamStoreWatchTimeout
                Messages:       Unexpected watchers number
```
This was because terminating watchers is asynchronous and it could take some extra time to receive the terminated notification. The extra time was set to 10 ms which turned out to be not safe enough as we have seen this failure.

**To Reproduce**
Benchmark the test hundreds of times, it may encounter the failure.
```
# go test  -run TestRamStoreWatchTimeout  -count=200 github.com/vmware-tanzu/antrea/pkg/apiserver/storage/ram
--- FAIL: TestRamStoreWatchTimeout (0.14s)
    store_test.go:509: w2 was not stopped, expected stopped
    store_test.go:511:
                Error Trace:    store_test.go:511
                Error:          Not equal:
                                expected: 1
                                actual  : 2
                Test:           TestRamStoreWatchTimeout
                Messages:       Unexpected watchers number
FAIL
FAIL    github.com/vmware-tanzu/antrea/pkg/apiserver/storage/ram        24.113s
FAIL
```

**Expected**
The test should succeed consistently when the implementation is correctly.

**Actual behavior**
It failed occasionally.

**Versions:**
Please provide the following information:
 - Antrea version (Docker image tag). ee1a0abc219c76c0d5e4d0d65cf0f718ffe61c33

--- a/pkg/apiserver/storage/ram/store_test.go
+++ b/pkg/apiserver/storage/ram/store_test.go
@@ -494,10 +494,12 @@ func TestRamStoreWatchTimeout(t *testing.T) {
 		store.Create(&v1.Pod{ObjectMeta: metav1.ObjectMeta{Name: fmt.Sprintf("pod%d", i), Labels: map[string]string{"app": "nginx"}}})
 	}
 
+	// Terminating watchers is asynchronous, leave it some reaction time to avoid flakes.
+	terminationReactionTime := time.Millisecond * 100
 	select {
 	case <-w2.(*storeWatcher).done:
 		t.Fatal("w2 was stopped, expected not stopped")
-	case <-time.After(watcherAddTimeout + time.Millisecond*10):
+	case <-time.After(watcherAddTimeout + terminationReactionTime):
 	}
 
 	// w2 can't take one more event as it's buffer has been full, it should be terminated.
@@ -505,7 +507,7 @@ func TestRamStoreWatchTimeout(t *testing.T) {
 
 	select {
 	case <-w2.(*storeWatcher).done:
-	case <-time.After(watcherAddTimeout + time.Millisecond*10):
+	case <-time.After(watcherAddTimeout + terminationReactionTime):
 		t.Error("w2 was not stopped, expected stopped")
 	}
 	assert.Equal(t, 1, store.GetWatchersNum(), "Unexpected watchers number")

